### DISCUSSION TITLE
Apache questions

### DISCUSSION THREAD

**Discussion by caphector** [aaronwmorris/indi-allsky#970] [Q&A] [answered]
I've set up my home LAN with signed certs so I don't have to worry about certificate errors. I'd edited the indi-allsky.conf to use my signed certs. I noticed that when I ran setup.py again it regenerated the config file with the stock certs; it's only a moment to fix it, but I wanted to mention it.

I was re-running the setup.py because I had moved my image storage to an external drive; I noticed that the image viewer now is failing to load some images. The `webapp-indi-allsky.log` has errors like:

`Oct  2 09:56:31 skycam [ERROR] MainProcess forms.getImages() #2528: Error determining relative file name: '/var/www/html/allsky/images/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230928/day/28_17/ccd1_20230928_170010.jpg' is not in the subpath of '/mnt/stars/allsky' OR one path is relative and the other is absolute.`.

I can `ls` the files so that's working and it only seems to be a subset of images.

**Comment by aaronwmorris** [comment]
For the certificates, I would just copy your certs to the location that indi-allsky expects and it will just use those when you re-run setup.sh

Cert:  `/etc/apache2/ssl/indi-allsky_apache.pem`
Key: `/etc/apache2/ssl/indi-allsky_apache.key`

As for the file locations, if you moved all of your files, the easiest thing you can probably do is just create a symbolic link to simulate the old path.  Something like this.

```
ln -s /mnt/stars/allsky/images /var/www/html/allsky/images
```

**Reply by caphector** [reply]
Renaming the certs is an easier solution; wish I'd thought of it.

I created a symlink which, oddly, seems to work for some images but not others. 

```
ls -l /var/www/html/allsky
lrwxrwxrwx 1 skycam skycam 18 Oct  1 19:38 images -> /mnt/stars/allsky/
```

All the files are immediately inside /mnt/stars/allsky/:

```
$ ls /mnt/stars/allsky/
ccd_54528347-52c5-49e9-9fe2-65c3a007035f  darks  export  js  latest.jpg
```

The same thing happens if I create a symlink to `/mnt/stars/allsky`:

```
$ ls -l
total 0
lrwxrwxrwx 1 skycam skycam 17 Oct  2 18:52 images -> /mnt/stars/allsky
```

What really confuses me here is that it takes a couple attempts before it fails; all the images are on the new drive but only some of them fail to load. I can `ls` all images at either path:

```
for i in $(tail -n 3 /var/log/indi-allsky/webapp-indi-allsky.log | cut -d \' -f2)
do
  ls $i ${i/\/var\/www\/html\/allsky\/images/\/mnt\/stars\/allsky}
done
/mnt/stars/allsky/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040042.jpg
/var/www/html/allsky/images/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040042.jpg
/mnt/stars/allsky/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040029.jpg
/var/www/html/allsky/images/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040029.jpg
/mnt/stars/allsky/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040011.jpg
/var/www/html/allsky/images/ccd_54528347-52c5-49e9-9fe2-65c3a007035f/20230930/night/01_04/ccd1_20231001_040011.jpg
```

**Reply by aaronwmorris** [reply]
Yeah, I realize this is a little more complex than I originally thought.  Let me think about this.

**Reply by caphector** [reply]
I am thinking of fixing the paths in the db. Is there a reason the db paths are absolute?


Example:
```
UPDATE darkframe
SET filename = REPLACE(filename, "/var/www/html/allsky/images", "/mnt/stars/allsky");
```

**Reply by aaronwmorris** [reply]
I _think_ the file paths can be relative, its something I have been hoping to test.  The full path is an artifact from older indi-allsky code. 
 But replacing the path should work just fine.

**Reply by caphector** [reply]
I updated my paths to be relative and it *seems* to be working fine so far. I updated DBs `video`, `keogram`, `darkframe`, `image`, `startrail`, and `startrailvideo` to use a relative path in place of the old path; I haven't changed all the paths yet. I notice that I still get a 500 error (due to the same underlying issue as per the logs) when I go to Tools -> Dark Library; not sure where that is pulling it's path from.

**Comment by aaronwmorris** [comment, marked as answer]
Merged #976 so that all file paths are now stored as relative paths in the database.  I have tried to track down all of the places where I was referencing the `.filename` attribute directly and replaced them with `.getFilesystemPath()` or `.getUrl()` (for the relative path).

**Reply by caphector** [reply]
Thank you for merging this; I've updated my DB to only use relative paths and it appears to be working properly; the last of the 500 errors stopped when I updated the filenames in the `badpixelmap` table.
